.s2i/bin/assemble

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+echo "--> Adding yarn to the image"
+npm install -g yarn
+
+set -e
+safeLogging () {
+    if [[ $1 =~ http[s]?://.*@.*$ ]]; then
+        echo $1 | sed 's/^.*@/redacted@/'
+    else
+        echo $1
+    fi
+}
+
+shopt -s dotglob
+if [ -d /tmp/artifacts ] && [ "$(ls /tmp/artifacts/ 2>/dev/null)" ]; then
+    echo "---> Restoring previous build artifacts ..."
+    mv -T --verbose /tmp/artifacts/node_modules "${HOME}/node_modules"
+fi
+
+echo "---> Installing application source ..."
+mv /tmp/src/* ./
+
+# Fix source directory permissions
+fix-permissions ./
+
+
+echo "---> Installing all dependencies"
+NODE_ENV=development yarn install
+
+#do not fail when there is no build script
+echo "---> Building in production mode"
+yarn run build
+
```

Comment:
You should clean up all your dev dependencies after build so the image is as small as possible.

Reply:
Changed it to only install production dependencies